backends/bmv2/bmv2.cpp

```diff
@@ -84,12 +84,26 @@ int main(int argc, char *const argv[]) {
         return 1;
 
     // backend depends on the modified refMap and typeMap from midEnd.
-    BMV2::Backend backend(options.isv1(), &midEnd.refMap,
-            &midEnd.typeMap, &midEnd.enumMap);
+    BMV2::Backend backend(options, &midEnd.refMap, &midEnd.typeMap, &midEnd.enumMap);
+
+    if (auto arch = getenv("P4C_DEFAULT_ARCH")) {
```

Comment:
This kind of pattern suggests to me that you actually have a hidden subtype hidden: you probably want a base class for backend with two subclasses for SS and PSA.